File: src/main/java/cz/astro/var/data/helper/BooleanToStringConverter.java
package cz.astro.var.data.helper;

import javax.persistence.AttributeConverter;
import javax.persistence.Converter;

/**
 * @author Michal
 * @version 1.0
 * @since 10/15/2018
 */
@Converter
public class BooleanToStringConverter implements AttributeConverter<Boolean, String> {

    @Override
    public String convertToDatabaseColumn(Boolean value) {
        return (value != null && value) ? "Y" : "N";
    }

    @Override
    public Boolean convertToEntityAttribute(String value) {
        return "Y".equals(value);
    }
}

File: backend/src/main/kotlin/cz/astro/var/data/helper/BooleanToStringConverter.kt
package cz.astro.`var`.data.helper

import javax.persistence.AttributeConverter
import javax.persistence.Converter

/**
 * @author Michal
 * @version 1.0
 * @since 10/15/2018
 */
@Converter
class BooleanToStringConverter : AttributeConverter<Boolean, String> {

    override fun convertToDatabaseColumn(value: Boolean?): String {
        return if (value != null && value) "Y" else "N"
    }

    override fun convertToEntityAttribute(value: String): Boolean? {
        return "Y" == value
    }
}
